backend/src/packages/chaiNNer_pytorch/pytorch/processing/align_image_to_reference.py

```diff
@@ -221,14 +221,19 @@ def align_image_to_reference_node(
     blur_strength: float,
 ) -> np.ndarray:
     context.add_cleanup(safe_cuda_cache_empty)
+    exec_options = get_settings(context)
     multiplier = precision.value / 1000
-    return align_images(
-        context,
-        target_img,
-        source_img,
-        precision,
-        multiplier=multiplier,
-        alignment_passes=alignment_passes,
-        blur_strength=blur_strength,
-        ensemble=True,
-    )
+    try:
+        return align_images(
+            context,
+            target_img,
+            source_img,
+            precision,
+            multiplier=multiplier,
+            alignment_passes=alignment_passes,
+            blur_strength=blur_strength,
+            ensemble=True,
+        )
+    finally:
+        if exec_options.force_cache_wipe:
+            safe_cuda_cache_empty()
```

Comment:
Maybe instead of doing this everywhere, we could make `context.add_cleanup(fn, after="node")` a thing. Then we could write all of this as just
```py
context.add_cleanup(safe_cuda_cache_empty, after="node" if exec_options.force_cache_wipe else "chain")
```

Reply:
That's a better idea for sure